syntax = "proto3";

package inference.v1;

option go_package = "runner/proto/client/inference/v1;inferencepb";

// Сервис для детекции объектов на изображениях
service InferenceService {
  // Метод для обработки одного изображения
  rpc Detect(DetectRequest) returns (DetectResponse);
}

// Запрос с изображением
message DetectRequest {
  bytes image = 1; // Изображение в байтах (JPEG, PNG и т.д.)
}

// Прямоугольник с координатами объекта
message Rectangle {
  float x0 = 1; // Левая верхняя точка X
  float y0 = 2; // Левая верхняя точка Y
  float x1 = 3; // Правая нижняя точка X
  float y1 = 4; // Правая нижняя точка Y
}

// Один обнаруженный объект
message Detection {
  string class_name = 1;  // Название класса объекта
  Rectangle rectangle = 2; // Координаты bounding box
}

// Ответ с результатами детекции
message DetectResponse {
  repeated Detection detections = 1; // Список обнаруженных объектов
}

